protocols/relay/src/behaviour.rs

```diff
@@ -253,27 +254,132 @@ pub struct Behaviour {
 
     local_peer_id: PeerId,
 
+    connections: HashMap<PeerId, HashSet<ConnectionId>>,
     reservations: HashMap<PeerId, HashSet<ConnectionId>>,
```

Comment:
Im almost inclined to say we should merge this in some way so we can track connections and the connections that have reservations. Could probably be done in a separate PR though. Thoughts?